drivers/sensor/maxim/ds3231/ds3231.c

```diff
@@ -74,13 +84,7 @@ static int sensor_ds3231_channel_get(const struct device *dev, enum sensor_chann
 
 	switch (chan) {
 	case SENSOR_CHAN_AMBIENT_TEMP:
-		const uint16_t raw_temp = data->raw_temp;
-
-		val->val1 = (int8_t)(raw_temp & GENMASK(8, 2)) >> 2;
-
-		uint8_t frac = raw_temp & 3;
-
-		val->val2 = (frac * 25) * pow(10, 4);
+		sensor_ds3231_temp_from_raw(data, val);
```

Comment:
You can wrap the block in curly braces instead of pulling out to a separate function.

Reply:
This has already been clarified, see: https://github.com/zephyrproject-rtos/zephyr/pull/85030#discussion_r1938898404

For MISRA rule reasons, statement blocks are considered “undesirable”, so there is now a static inline function for this (also my preferred variant).